### ISSUE TITLE
Button keymaps no longer cancel out predefined actions

### ISSUE THREAD

**Issue by barcoboy** [sds100/KeyMapper#418]
Hi there.  As others have said, this is an excellent app, and I thank you for writing it.

I'm having a problem with the newest version in that when I define a keymap for a physical button, it no longer cancels out the predefined Android action that is assigned to that same button.  For example, I want to use Key Mapper to open an application when I long press the Home button.  I define the trigger and action in Key Mapper, and when I long press the Home button, it does start the application I have chosen, but my phone also starts the Google Assistant app listening for a voice command.  This one I can work around by turning off the Google Assistant, but if I for example define an action to a long press of one of the volume buttons, again the action I have defined works, but the phone's volume also gets adjusted.

Running version 2.0.0 on two identical rooted Samsung Galaxy S5 phones with Android 6.0.1.

**Comment by sds100** [comment]
Hello,

does this happen with all the buttons on your device? Volume down, up, home, back and recents.

**Comment by sds100** [comment]
Have you tried rebooting?

**Comment by barcoboy** [comment]
Well it seems that the problem has resolved itself with the volume buttons, but is still present with the Home button 100% of the time.  The Back and Recent buttons seem to work okay.  I have rebooted multiple times, cleared all data files, and uninstalled and re-installed.

**Comment by barcoboy** [comment]
OK, I figured out what happened with the volume buttons.  In testing just now, I set a keymap to toggle the flashlight when the Volume Down button is long pressed, but accidentally forgot to set the long press option.  So when I long pressed the Volume Down key, the flashlight would toggle, but the volume control would either step down one step, or drop right down to zero.  Setting the long press option in the keymap fixed the action so that a long press would toggle the flashlight, and a short press would adjust the volume one step.  But setting the option back to short press and testing the action with a short press of the button causes the flashlight to toggle and the volume to drop a step or drop down to zero as before.  Rebooting causes no change.

**Comment by sds100** [comment]
Hi, I think I made some very subtle changes from the last version so I'm going to see what happened.

**Comment by sds100** [comment]
@barcoboy Could you try this apk to see if it fixes your problem...
[keymapper-2.0.0-debug.zip](https://github.com/sds100/KeyMapper/files/5051167/keymapper-2.0.0-debug.zip)

**Comment by sds100** [comment]
Is your last comment about the volume buttons a bug?

**Comment by barcoboy** [comment]
> @barcoboy Could you try this apk to see if it fixes your problem...
> [keymapper-2.0.0-debug.zip](https://github.com/sds100/KeyMapper/files/5051167/keymapper-2.0.0-debug.zip)

Happy to report that this debug version does fix the Home button problem, and the Google Assistant no longer appears when I assign a keymap to Home longpress.  :-)

> Is your last comment about the volume buttons a bug?

Yes this is still occurring.  If I assign the Toggle Flashlight action to a Volume button short press, pressing the Volume button both toggles the flashlight and lowers the volume.  Interestingly though this only happens when the flashlight is being turned on, and does not happen when the flashlight is being turned off.  More testing shows that other actions, including opening apps, app shortcuts, or other functions such as Toggle WiFi or Toggle Airplane Mode, all work fine and do not change the volume.  

An even more interesting result is I assigned Volume Up to turn on the flashlight, and Volume Down to turn it off.  Pressing Volume Up when the flashlight was off turned it on, and also increased my volume one notch.  BUT, pressing Volume Up again while the flashlight was already on did not increase my volume any further, no matter how many times the button was pressed.  It was only after I pressed Volume Down to turn the flashlight off and then pressed Volume Up to turn it back on again was the volume increased another notch.  Also, pressing Volume Down no matter what state the flashlight was in never decreased the volume.  So it seems that the action of actually turning on the flashlight is the cause, whether it is done by a "Toggle Flashlight" or "Enable Flashlight" action.  I hope this all makes sense.

**Comment by sds100** [comment]
Hi, that is a really strange and specific bug so I have no idea why Key Mapper would cause that. I also can't reproduce that behaviour. Key Mapper doesn't treat the flashlight actions differently to any of the other actions you tested so they _should_ also cause this problem. And this _only_ happens in the latest update? Can you double check it didn't happen in 1.1.7? you can get that apk from the releases section in the repo.

**Comment by barcoboy** [comment]
I just tried 1.1.7, and the same thing occurs.  It's not really an issue for me and I don't think it would be for many people, as all of the follow have to be true for it to trigger:

- A short press volume trigger needs to be defined (overriding the volume adjustment)
- A flashlight enable/toggle action needs to be defined
- Running on an affected piece of hardware, possibly Samsung or even Galaxy S5 specifically.

I think most people (as is the case for me) when assigning actions to the volume buttons would assign them to a long press, so this would not occur.  And it's only the volume being increased by a notch and the volume level on screen display popping up for a second; it's not like the app is crashing or data is being corrupted.

Just to make sure it wasn't my one specific phone, I tried version 2.0.0 on my other Galaxy S5, and the issue occurs there too.  I wish I had more Android devices I could test this on for you.
